shell/platform/windows/uwptool_main.cc

```diff
@@ -21,27 +21,46 @@ namespace {
 void PrintInstalledApps() {
   flutter::ApplicationStore app_store;
   for (const flutter::Application& app : app_store.GetInstalledApplications()) {
-    std::wcout << app.GetPackageId() << std::endl;
+    std::wcout << app.GetPackageFamily() << std::endl;
   }
 }
 
-// Launches the app installed on the system whose Application User Model ID is
-// prefixed with app_id, with the specified arguments list.
+// Launches the app installed on the system with the specified package family.
 //
 // Returns -1 if no matching app, or multiple matching apps are found, or if
 // the app fails to launch. Otherwise, the process ID of the launched app is
 // returned.
-int LaunchApp(const std::wstring_view app_id, const std::wstring_view args) {
-  flutter::Application app(app_id);
-  return app.Launch(args);
+int LaunchApp(const std::wstring_view package_family,
+              const std::wstring_view args) {
+  flutter::ApplicationStore app_store;
+  std::optional<flutter::Application> app =
+      app_store.GetInstalledApplication(package_family);
+  if (!app) {
+    return -1;
+  }
+  return app->Launch(args);
+}
+
+// Uninstalls the app with the specified package family.
+//
+// Returns true on success.
+bool UninstallApp(const std::wstring_view package_family) {
+  flutter::ApplicationStore app_store;
+  std::optional<flutter::Application> app =
+      app_store.GetInstalledApplication(package_family);
+  if (!app) {
+    return false;
+  }
+  return app->Uninstall();
 }
 
 // Prints the command usage to stderr.
 void PrintUsage() {
-  std::cerr << "usage: uwptool COMMAND [APP_ID]" << std::endl;
```

Comment:
I'm wondering if we might need to keep both overloads?  In the future, when flutter tool runs `flutter create` it will generate a name presumably derived from the flutter project name and inject that into the appx manifest template  (line 8 here https://github.com/clarkezone/runner-uwp/blob/dev/src/winuwp/runner_uwp/appxmanifest.in) prior to calling cmake.

Later when it calls the build machinery, a build and package will be created derived from that.  After the build has happened flutter tools will want to install and launch.  At that point, I would guess it will want to use the name it generated / derived from the dart project.

Another interesting case is:

- User runs `flutter create`
- User builds package, iterates
- User opens the `.sln` in visual studio and associates the app with a Windows Store application to get ready to publish

The above will change the package name / identity.  At that point `fluuter run` might need to be able to call run using the name.  To support this flow the flutter tool would retrieve the Name from the xml manifest and use that in subsequent tool invocations.  Arguably the tool should always parse the manifest prior to calling the tool rather than remember the name it allocated.

Reply:
Good point. It sounds like what we probably need is to wire up a command that allows retrieval of package info from the app GUID, then either always work from there, or use it as needed.

Makes sense. Would you find filing a bug with an example showing these steps and what the App ID and package ID look like after each one? Specifically the before and after of when the user associates the app with a windows store app just before publishing.